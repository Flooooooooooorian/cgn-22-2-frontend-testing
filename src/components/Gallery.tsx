import {Item} from "../Item";
import ItemCard from "./ItemCard";

type GalleryProps = {
    items: Item[]
}

export default function Gallery(props: GalleryProps) {


    return (
        <>
            <h1>Item-Gallery</h1>
            {props.items.map((item) => <ItemCard key={item.id} item={item}/>)}
        </>
    )
}
