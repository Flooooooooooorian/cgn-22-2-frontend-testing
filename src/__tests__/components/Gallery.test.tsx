import {Item} from "../../Item";
import {render} from "@testing-library/react";
import Gallery from "../../components/Gallery";

jest.mock("../../components/ItemCard", () => {
    return (props: any) => {
        return <div {...props}></div>
    }
})

describe("Gallery test", () => {

    test("Gallery Snapshot test", () => {
        //GIVEN
        const testItems: Item[] = [{id: "testId", name: "testName"}, {id: "testId2", name: "testName2"}];

        //WHEN
        const view = render(<Gallery items={testItems}/>)

        //THEN
        expect(view.asFragment()).toMatchSnapshot();
    })
})
