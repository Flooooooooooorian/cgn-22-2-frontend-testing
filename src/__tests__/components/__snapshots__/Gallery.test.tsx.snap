// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery test Gallery Snapshot test 1`] = `
<DocumentFragment>
  <h1>
    Item-Gallery
  </h1>
  <div
    item="[object Object]"
  />
  <div
    item="[object Object]"
  />
</DocumentFragment>
`;
